pyscriptjs/src/utils.ts

```diff
@@ -84,4 +86,105 @@ function handleFetchError(e: Error, singleFile: string) {
     showError(errorContent);
 }
 
-export { addClasses, removeClasses, getLastPath, ltrim, htmlDecode, guidGenerator, showError, handleFetchError };
+function readTextFromPath(path: string) {
+    const request = new XMLHttpRequest();
+    request.open("GET", path, false);
+    request.send();
+    const returnValue = request.responseText;
+
+    return returnValue;
+}
+
+function inJest(): boolean {
+    return typeof process === 'object' && process.env.JEST_WORKER_ID !== undefined;
+}
+
+function mergeConfig(inlineConfig: AppConfig, externalConfig: AppConfig): AppConfig {
+    const name = inlineConfig.name || externalConfig.name;
+    const description = inlineConfig.description || externalConfig.description;
+    const version = inlineConfig.version || externalConfig.version;
+    const type = inlineConfig.type || externalConfig.type;
+    const author_name = inlineConfig.author_name || externalConfig.author_name;
+    const author_email = inlineConfig.author_email || externalConfig.author_email;
+    const license = inlineConfig.license || externalConfig.license;
+    const autoclose_loader = (typeof inlineConfig.autoclose_loader !== "undefined") ? inlineConfig.autoclose_loader : externalConfig.autoclose_loader;
+    const runtimes = inlineConfig.runtimes || externalConfig.runtimes;
+    const packages = inlineConfig.packages || externalConfig.packages;
+    const paths = inlineConfig.paths || externalConfig.paths;
+    const plugins = inlineConfig.plugins || externalConfig.plugins;
+    const merged: AppConfig = {
+        name,
+        description,
+        version,
+        type,
+        author_name,
+        author_email,
+        license,
+        autoclose_loader,
+        runtimes,
+        packages,
+        paths,
+        plugins
+    };
+
+    return merged;
+}
+
+function validateConfig(configText: string) {
+    let config: object;
+    try {
+        config = JSON.parse(configText);
+    }
+    catch (err) {
+        console.log(err.message);
+    }
+
+    const finalConfig: AppConfig = {}
+    const allKeys = {
+        "string": ["name", "description", "version", "type", "author_name", "author_email", "license"],
+        "boolean": ["autoclose_loader"],
+        "array": ["runtimes", "packages", "paths", "plugins"]
```

Comment:
Good to have a test that demonstrates packages takes paths as well, default runtimes paths.

Reply:
what does `packages takes paths as well, default runtimes paths.` mean? I thought it only took stuff like `["numpy", "pandas"]`. That's why we have a separate key for `"paths"` which can take stuff like `["./utils.py"]`

Reply:
@fpliger same understanding as madhur's? Then we are good.

Reply:
resolving this